src/com/android/volley/toolbox/ImageRequest.java

```diff
@@ -141,8 +141,11 @@ private static int getResizedDimension(int maxPrimary, int maxSecondary, int act
                     return Response.error(new ParseError(response));
                 } else {
                     Cache.Entry entry = HttpHeaderParser.parseCacheHeaders(response);
-                    if (entry != null) {
+                    if (entry != null && getTTL() > 0) {
                         entry.setTTL(System.currentTimeMillis() + getTTL());
+                        if (isOfflineCache()) {
+                            entry.keepUntil = entry.ttl;
```

Comment:
so my change modifies the ttl to be the cache until ttl. Because there is a softttl from the server api. so this works as is.